java/Lexer.java

```diff
@@ -1,9 +1,12 @@
 import java.util.HashMap;
+import java.util.Optional;
 
 class Lexer {
-    String input;
-    int pos;
-    HashMap<String, Token> keywordMap = new HashMap<String, Token>() {
+
+    private final String input;
+    private int pos = 0;
```

Comment:
Is there a reason to make all values final?

Reply:
I believe it's a best practice to make all values that shouldn't change final to avoid unnecessary mutation and hacks, like is the default in many other languages. I don't have a strong preference for this, but I got used to putting final on everything since it's my company's default.

Reply:
I removed final from local variables.